systemtest/src/main/java/io/strimzi/systemtest/utils/kafkaUtils/KafkaUtils.java

```diff
@@ -68,7 +68,11 @@ public static void waitForKafkaNotReady(String clusterName) {
     }
 
     public static void waitForKafkaStatus(String clusterName, Enum<?>  state) {
-        Kafka kafka = KafkaResource.kafkaClient().inNamespace(kubeClient().getNamespace()).withName(clusterName).get();
+        String namespace = kubeClient().getNamespace();
+        Kafka kafka = KafkaResource.kafkaClient().inNamespace(namespace).withName(clusterName).get();
+        if (kafka == null) {
+            throw new RuntimeException(String.format("Kafka resource %s not found in namespace %s", clusterName, namespace));
+        }
```

Comment:
If `waitForKafkaStatus` throwed NPE, this will throw exception on the same place and the test will end here.
Maybe it would be great to wait for the Kafka (re)creation and then `waitForKafkaStatus`.

Something like this:
```
    public static void waitForKafkaRecreation(String clusterName) {
        LOGGER.info("Waiting for recreation of Kafka:{}", clusterName);
        TestUtils.waitFor("Kafka recreation " + clusterName, Constants.POLL_INTERVAL_FOR_RESOURCE_READINESS, Constants.GLOBAL_TIMEOUT,
            () -> KafkaResource.kafkaClient().inNamespace(kubeClient().getNamespace()).withName(clusterName).get() != null);

        LOGGER.info("Kafka: {} recreated.", clusterName);
    }
```
and you can add it to `KafkaUtils`.
Then you'll do just
```
KafkaUtils.waitForKafkaRecreation(clusterName);
KafkaUtils.waitForKafkaReady(clusterName);
```

Reply:
The ST is still failing at the same point on Azure, this time is captured by my `RuntimeException`. You may be right, but I'm wondering why this never happens on my local env. Anyway, thanks! I'll review later.

### Mine
```log
2021-03-17 15:58:38 INFO  [Exec:172] ======STDOUT END======
2021-03-17 15:58:38 INFO  [ColdBackupScriptST:88] Waiting for provisioning of restored cold-backup/my-cluster-1300602412
2021-03-17 15:58:38 DEBUG [CustomResource:157] Calling CustomResource#setKind doesn't do anything because the Kind is computed and shouldn't be changed
2021-03-17 15:58:38 INFO  [ResourceManager:449] Wait for Kafka: my-cluster-1300602412 will have desired state: Ready
...
2021-03-17 15:59:05 INFO  [TimeMeasuringSystem:113] End time of operation TEST_EXECUTION is correctly stored
2021-03-17 15:59:05 INFO  [AbstractST:655] Search in strimzi-cluster-operator log for errors in last 442 seconds
2021-03-17 15:59:06 INFO  [BaseCmdKubeClient:439] Exception not found
```

### Azure
```log
2021-03-17 16:35:47 INFO  [Exec:172] ======STDOUT END======
2021-03-17 16:35:47 INFO  [ColdBackupScriptST:88] Waiting for provisioning of restored cold-backup/my-cluster-900682292
2021-03-17 16:35:48 INFO  [TimeMeasuringSystem:113] End time of operation TEST_EXECUTION is correctly stored
...
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 745.473 s <<< FAILURE! - in io.strimzi.systemtest.backup.ColdBackupScriptST
[ERROR] io.strimzi.systemtest.backup.ColdBackupScriptST.backupAndRestore  Time elapsed: 733.172 s  <<< ERROR!
java.lang.RuntimeException: Kafka resource my-cluster-900682292 not found in namespace cold-backup
	at io.strimzi.systemtest.utils.kafkaUtils.KafkaUtils.waitForKafkaStatus(KafkaUtils.java:74)
	at io.strimzi.systemtest.utils.kafkaUtils.KafkaUtils.waitForKafkaReady(KafkaUtils.java:63)
	at io.strimzi.systemtest.backup.ColdBackupScriptST.backupAndRestore(ColdBackupScriptST.java:89)
```

Reply:
The problem is, that on the other env it can be somehow slower than on your machine, so these things are causing race conditions. When you add the wait there and revert changes in `waitForKafkaStatus`, it will be okay.

Reply:
@im-konge I'm not able to reproduce the issue on my end probably because of different timing. 

Just committed your suggestion, let's see if it works now.

Reply:
Okay, if it will work, revert please changes in `waitForKafkaStatus`. Thanks! :)

Reply:
ok

Reply:
So the test failed again, because Kafka is not present in 5 minutes. Is it possible that is required k8s version `1.20.0`+? On Azure we are running tests on k8s `1.16.0`, so I triggered Jenkins pipeline (where is `1.20+`) to see if it will have any effect.

Reply:
Ok, let me know. Have you tried to run it locally?